torch/distributed/distributed_c10d.py

```diff
@@ -520,6 +520,10 @@ def init_process_group(backend,
             store, rank, world_size = next(rendezvous_iterator)
             store.set_timeout(timeout)
 
+            # Use a PrefixStore to avoid accidental overrides of keys used by
+            # different systems (e.g. RPC) in case the store is multi-tenant.
+            store = PrefixStore("default_pg", store)
```

Comment:
Since multitenancy is not introduced in this PR, does it make more sense for this change to go in the 4th PR of the stack? Although it seems fine here too as long as it doesn't create any issues.

Reply:
See my comment below.